### DISCUSSION TITLE
Initialize Firebase without GoogleService-Info.plist

### DISCUSSION THREAD

**Discussion by flying-turtle-tech** [firebase/firebase-ios-sdk#8603] [Q&A] [answered]
Hi,
I'm wondering if it's possible to initialize Firebase without needing the GoogleService plist. If not maybe a feature could be added where all the info could be stored in a struct and provided during initialization? 

I am asking because the app I am working on has security as it's highest priority, so we obfuscate everything. Sometimes we aren't able to obfuscate files so it would be helpful to be able to add the info as a struct, and obfuscate it in code, rather than a file on disk.

**Comment by ncooke3** [comment, marked as answer]
Hi there and thanks for the question. 

Have you considered the [`FirebaseApp.configure(options:)`](https://firebase.google.com/docs/reference/swift/firebasecore/api/reference/Classes/FirebaseApp#configureoptions:) API? Essentially, you can create a `FirebaseOptions` object, set the configuration info (from your`GoogleService-Info.plist`) and then pass that into the above API to configure the Firebase App. This prevents the need for including a `GoogleService-Info.plist` file in your project.

**Reply by paulb777** [reply]
@ncooke3's answer should work for most of Firebase but not for Firebase Analytics - See #230
